File: app/src/main/java/com/example/mobv2/ui/callbacks/SetAddressCallback.java
package com.example.mobv2.ui.callbacks;

import android.content.Context;
import android.util.Log;
import android.widget.Toast;

import com.example.mobv2.serverapi.MOBServerAPI;
import com.example.mobv2.ui.activities.MainActivity;
import com.google.gson.internal.LinkedTreeMap;

public class SetAddressCallback implements MOBServerAPI.MOBAPICallback
{
    private final Context context;

    public SetAddressCallback(Context context)
    {
        this.context = context;
    }

    @Override
    public void funcOk(LinkedTreeMap<String, Object> obj)
    {
        Log.v("DEBUG", obj.toString());
        LinkedTreeMap<String, String> response =
                (LinkedTreeMap<String, String>) obj.get("response");
        Toast.makeText(context, "Successful", Toast.LENGTH_LONG)
             .show();

        MainActivity.token = response.get("token");

    }

    @Override
    public void funcBad(LinkedTreeMap<String, Object> obj)
    {
        Log.v("DEBUG", obj.toString());
    }

    @Override
    public void fail(Throwable obj)
    {
        Log.v("DEBUG", obj.toString());
    }
}

File: app/src/main/java/com/example/mobv2/callback/SetAddressCallback.kt
package com.example.mobv2.callback

import android.util.Log
import com.example.mobv2.ui.activity.mainActivity.MainActivity
import com.google.gson.internal.LinkedTreeMap
import serverAPI.MOBServerAPI.MOBAPICallback

class SetAddressCallback(private val mainActivity: MainActivity) : MOBAPICallback
{
    override fun funcOk(obj: LinkedTreeMap<String, Any>)
    {
        Log.v("DEBUG", obj.toString())
        val response = obj["response"] as LinkedTreeMap<String, String>?

        MainActivity.token = response!!["token"]
    }

    override fun funcBad(obj: LinkedTreeMap<String, Any>)
    {
        Log.v("DEBUG", obj.toString())
    }

    override fun fail(obj: Throwable)
    {
        Log.v("DEBUG", obj.toString())
    }
}